pkg/services/ngalert/api/tooling/definitions/alertmanager.go

```diff
@@ -259,6 +276,17 @@ func (c *PostableUserConfig) EncryptSecureSettings() error {
 					}
 					gr.SecureSettings[k] = base64.StdEncoding.EncodeToString(encryptedData)
 				}
+				if gr.Uid == "" {
+					for {
+						gen := util.GenerateShortUID()
```

Comment:
Seems like it should count and stop after a number of retries, or only try once.